pkg/controller/daemon/controller.go

```diff
@@ -370,6 +373,47 @@ func (dsc *DaemonSetsController) getNodesToDaemonPods(ds *extensions.DaemonSet)
 	return nodeToDaemonPods, nil
 }
 
+func (dsc *DaemonSetsController) needsRollingUpdate(ds *extensions.DaemonSet, daemonPods []*api.Pod, numUnavailable, numNodes int) (*api.Pod, bool) {
+	maxUnavailable, isPercent, err := util.GetIntOrPercentValue(&ds.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable)
+	if err != nil {
+		glog.Errorf("Invalid value for MaxUnavailable: %v", err)
+		return nil, false
+	}
+	if isPercent {
+		maxUnavailable = util.GetValueFromPercent(maxUnavailable, numNodes)
+	}
+	if numUnavailable >= maxUnavailable {
```

Comment:
Note that this has the unfortunate consequence of essentially permanently preventing updates if this threshold is reached due to crash loops.

See #16737.